exercises/practice/secret-handshake/.approaches/iterate-once/content.md

```diff
@@ -0,0 +1,101 @@
+# Iterate once
+
+**secret_handshake.h**
+```c
+#ifndef SECRET_HANDSHAKE_H
+#define SECRET_HANDSHAKE_H
+
+#include <stddef.h>
+
+const char **commands(size_t number);
+
+#endif
+```
+
+**secret_handshake.c**
+
+```c
+#include "secret_handshake.h"
+#include <malloc.h>
+
+static const char * signals[] = { "wink", "double blink", "close your eyes", "jump" };
+
+const char **commands(size_t n) {
+    const char **output = calloc(4, sizeof(char *));
+    int signal = 0, signal_incr = 1, output_idx = 0, end = 4;
+    if (n & 16) {signal = 3; signal_incr = -1; end = -1;}
+
+    for (; signal != end; signal+=signal_incr)
+        if (n & 1 << signal)
+            output[output_idx++] = signals[signal];
+    return output;
+}
+```
+
+This approach starts by defining an array of pointers to `const` `char`s.
```

Comment:
Question: would it be more accurate to say " pointers to `const char' strings"?

Maybe it's a linguistic thing.
There is an array that has multiple pointers each one to a string of const chars.

Reply:
Changed to

`This approach starts by defining an array of pointers to `const` `char` strings.`